dot/utils_test.go

```diff
@@ -0,0 +1,586 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package dot
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	ctoml "github.com/ChainSafe/gossamer/dot/config/toml"
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/internal/log"
+	"github.com/ChainSafe/gossamer/lib/genesis"
+	"github.com/ChainSafe/gossamer/lib/utils"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// newTestConfigWithFile returns a new test configuration and a temporary configuration file
+func newTestConfigWithFile(t *testing.T) (*Config, *os.File) {
+	cfg := NewTestConfig(t)
+
+	configPath := filepath.Join(cfg.Global.BasePath, "config.toml")
+	err := os.WriteFile(configPath, nil, os.ModePerm)
+	require.NoError(t, err)
+
+	cfgFile := exportConfig(cfg, configPath)
+	return cfg, cfgFile
+}
+
+// newTestGenesisFile returns a human-readable test genesis file using "gssmr" human readable data
+func newTestGenesisFile(t *testing.T, cfg *Config) (filename string) {
+	fp, err := utils.GetGssmrGenesisPath()
+	require.NoError(t, err)
+
+	gssmrGen, err := genesis.NewGenesisFromJSON(fp, 0)
+	require.NoError(t, err)
+
+	gen := &genesis.Genesis{
+		Name:       cfg.Global.Name,
+		ID:         cfg.Global.ID,
+		Bootnodes:  cfg.Network.Bootnodes,
+		ProtocolID: cfg.Network.ProtocolID,
+		Genesis:    gssmrGen.GenesisFields(),
+	}
+
+	b, err := json.Marshal(gen)
+	require.NoError(t, err)
+
+	filename = filepath.Join(t.TempDir(), "genesis.json")
+	err = os.WriteFile(filename, b, os.ModePerm)
+	require.NoError(t, err)
+
+	return filename
+}
+
+func TestCreateJSONRawFile(t *testing.T) {
+	type args struct {
+		bs *BuildSpec
+		fp string
+	}
+	tests := []struct {
+		name string
+		args args
+		want *os.File
```

Comment:
Change this to `expectedData string` and compare it with reading the file once it's returned?

Reply:
Changed to expectedHash and compared the sha256 hash of the file.